signer/src/transaction_coordinator.rs

```diff
@@ -721,28 +818,30 @@ pub fn given_key_is_coordinator(
     pub_key: PublicKey,
     bitcoin_chain_tip: &model::BitcoinBlockHash,
     signer_public_keys: &BTreeSet<PublicKey>,
-) -> Result<bool, Error> {
-    Ok(
-        coordinator_public_key(bitcoin_chain_tip, signer_public_keys)?
-            .map(|coordinator_pub_key| coordinator_pub_key == pub_key)
-            .unwrap_or(false),
-    )
+) -> bool {
+    coordinator_public_key(bitcoin_chain_tip, signer_public_keys) == Some(pub_key)
 }
 
 /// Find the coordinator public key
 pub fn coordinator_public_key(
     bitcoin_chain_tip: &model::BitcoinBlockHash,
     signer_public_keys: &BTreeSet<PublicKey>,
-) -> Result<Option<PublicKey>, Error> {
+) -> Option<PublicKey> {
```

Comment:
So what's the intended outcome if this would actually return `None`? Can the signer even function if that were the case? I'm thinking like, if other signers see this signer as the coordinator, but itself doesn't, then signing for the current block will be dead.

I guess I'm asking, should this be something-or-fatal?

Reply:
I believe this is something that is fatal. This should only return `None` if the `signer_public_keys` is an empty set, and that should never happen. If this does end up happening then that means there is some logic error in our code (right now even an empty bootstrap signing set in the config doesn't trigger this).

Reply:
Nothing to do now here and now imo, but if it is truly fatal & unrecoverable then it might deserve an `expect()` so the signer operator knows something is truly wrong (if it ever manages to happen).